src/ynViewer.sol

```diff
@@ -23,7 +26,40 @@ contract ynViewer is IynViewer {
     }
 
     /// @inheritdoc IynViewer
-    function getAllStakingNodes() public view returns (IStakingNode[] memory) {
-        return stakingNodesManager.getAllNodes();
+    function getRate() external view returns (uint256) {
+        uint256 _totalSupply = ynETH.totalSupply();
+        uint256 _totalAssets = ynETH.totalAssets();
+        if (_totalSupply == 0 || _totalAssets == 0) return 0;
```

Comment:
If any of these are 0 at the initial system state, the rate is actually 1.

Since ynETH:ETH starts as 1:1 and slowly decreases over time with appreciation.

Reply:
https://github.com/yieldnest/yieldnest-protocol/pull/119/commits/b41199f4c0257f6ec4bf6ed456f96f56d81e1dd5